### DISCUSSION TITLE
Why is the slider lagging behind the mouse?

### DISCUSSION THREAD

**Discussion by flosse** [vizia/vizia#448] [Questions]
When I compile the slider example `examples/views/slider.rs` (in release mode, of course), the slider is quite sluggish.
When I compile it with the `baseview` feature, it's just as you'd expect.
Why is that? Is there a latency between the emitting of the change event and the handling/update of the model?
With such a simple graphic, it's probably not the rendering, is it?

My system: Linux, Wayland, i7

**Comment by geom3trik** [comment]
Hmmm that is strange. I'm not sure why the baseview backend would work and not the winit one. The mouse won't perfectly stick to the slider thumb due to the double buffered nature of the opengl rendering but for such a small example and in release mode I wouldn't expect you to notice. I'll try to look into this but I can't replicate this issue currently and I don't have a Linux machine. Which suggests that it might be something to do with winit on wayland.

**Reply by flosse** [reply]
@geom3trik Thank you for your quick reply. I've tested it on three different systems now and the overall behavior seems to be the same. It looks like it is at least independent of the hardware. Even different operating systems (Mac/Linux) and ages (current vs. 18yrs old) don't make any difference.

**Reply by flosse** [reply]
In order to move from a feeling to reliable numbers, I measured the time that elapses between an `emit` and a processing (`AppData::value.map`) of the changed value: It sometimes takes up to **538 milliseconds**, whereas the distance between the `emit` and `event.map` the only takes ~**1.5µs** :thinking: But maybe I'm measuring completely wrong :see_no_evil:

**Reply by geom3trik** [reply]
Right, that's very strange that this is happening on multiple systems but not for everyone. Out of curiosity, does it improve at all if you comment out the labels? So it's just the sliders.

**Reply by flosse** [reply]
No, unfortunately not, it's exactly the same :disappointed:

**Reply by geom3trik** [reply]
Do any of the other examples lag like this? I'm thinking maybe the scrollview example when manipulating the scrollbars or the knob example.

**Reply by flosse** [reply]
Yes, it's exactly the same with the scrollview example. It's harder to see with the knob example, but i can't see any delay there so far.

**Reply by geom3trik** [reply]
This has completely perplexed me. The only thing I can think of is that something is going wrong with the mouse move events from winit. On all the systems you tested on did you use the same mouse? If so what kind of mouse is it?

**Reply by flosse** [reply]
Good point, I've always used the same wireless mouse, but unfortunately it's the same with a wired USB mouse :disappointed:

**Reply by uklotzde** [reply]
The `long_list` example is extremely slow. And 1000 items are not more than a few. I wouldn't expect that.

**Reply by geom3trik** [reply]
> Good point, I've always used the same wireless mouse, but unfortunately it's the same with a wired USB mouse :disappointed: 
> 

😞

Okay, maybe when I add more debugging tools to vizia it will become easier to figure this out. Unfortunately until I can replicate this I'm not sure there's much I can do. I'll try it on some other computers I have tomorrow and see what happens.

**Reply by geom3trik** [reply]
> The `long_list` example is extremely slow. And 1000 items are not more than a few. I wouldn't expect that.

There's some optimisations on the `views-update` branch that should fix this. Long story short, styling was much slower than it should be. If you found vizia via the reddit post then I apologise because vizia isn't quite ready for prime time, but it's very close.

**Reply by flosse** [reply]
I don't want to compare apples and oranges here, but with egui there is no such problem. Everything runs smoothly even on the 18 year old centrino.
Please don't take it the wrong way, I don't want to make anything bad here, on the contrary, because I think Vizia is so incredibly good, I want to contribute to making it even better :relaxed: 
Thanks in any case for all the work so far and the great framework!

**Reply by geom3trik** [reply]
> I don't want to compare apples and oranges here, but with egui there is no such problem. Everything runs smoothly even on the 18 year old centrino.
> Please don't take it the wrong way, I don't want to make anything bad here, on the contrary, because I think Vizia is so incredibly good, I want to contribute to making it even better :relaxed: 
> Thanks in any case for all the work so far and the great framework!

No I totally understand, a GUI framework shouldn't be laggy for such a simple example, and I'm sorry it's not working correctly. Just one last attempt, by any chance is the slider example on the `views-update` branch any better?

**Reply by flosse** [reply]
I am very sorry, but unfortunately there is no change to be seen there either. I have also removed all labels and the example page, but the problem still persists (see video)

[Bildschirmaufzeichnung vom 2024-01-28, 01-17-39.webm](https://github.com/vizia/vizia/assets/276043/9d3a863c-e93f-46bc-b0f2-152b40e4a82a)

**Comment by uklotzde** [comment]
I cannot confirm this issue on Fedora 39 on Wayland.

`cargo run --no-default-features --features wayland,winit --example slider`

No noticeable latency, even in debug mode.[^1]

[^1]: On an ancient Intel Skylake machine.

**Comment by uklotzde** [comment]
[Screencast from 2024-01-27 17-22-37.webm](https://github.com/vizia/vizia/assets/1474154/74cfdb8b-a912-47d0-8381-8799352bd382)

**Comment by flosse** [comment]
Here is how it looks like on my machine:
[Bildschirmaufzeichnung vom 2024-01-27, 17-57-50.webm](https://github.com/vizia/vizia/assets/276043/a4f2214c-c0e8-48f4-b69e-dd5bf26ed609)

And here with `baseview`
[Bildschirmaufzeichnung vom 2024-01-27, 17-59-33.webm](https://github.com/vizia/vizia/assets/276043/1af8056d-6399-46fc-ae33-f51b8560a02e)

*edit*:
It feels very similar on a Mac mini, i5, Intel HD 4000

*edit 2*:
It's the same on an old 32 Bit Intel Centrino with Linux and X11
